src/main/java/org/opensearch/knn/training/TrainingDataConsumer.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.training;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.opensearch.knn.index.JNIService;
+import org.opensearch.knn.index.memory.NativeMemoryAllocation;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * Transfers vectors from JVM to native memory.
+ */
+public class TrainingDataConsumer implements Consumer<List<Float[]>> {
+
+    private final NativeMemoryAllocation.TrainingDataAllocation trainingDataAllocation;
+
+    /**
+     * Constructor
+     *
+     * @param trainingDataAllocation NativeMemoryAllocation that contains information about native memory allocation.
+     */
+    public TrainingDataConsumer(NativeMemoryAllocation.TrainingDataAllocation trainingDataAllocation) {
+        this.trainingDataAllocation = trainingDataAllocation;
+    }
+
+    @Override
+    public void accept(List<Float[]> floats) {
+        trainingDataAllocation.setPointer(JNIService.transferVectors(trainingDataAllocation.getPointer(),
```

Comment:
So, I want all allocations to be created in the NativeMemoryLoadStrategy so that they can be properly tracked in the cache. That being said, I think I should make the Allocation constructors package private. What do you think?

Reply:
yes, make sense